app/mobile/lib/view/topic/long_press_dialog.dart

```diff
@@ -0,0 +1,208 @@
+                    fontWeight: FontWeight.bold),
+                textAlign: TextAlign.center,
+              ),
+              decoration: BoxDecoration(
+                  color: Color.fromARGB(255, 68, 84, 141),
+                  borderRadius: BorderRadius.circular(10)),
+            ),
+            GestureDetector(
+              onTap: () {
+                PersistentNavBarNavigator.pushNewScreen(context,
+                    screen: editResourceView(r));
+              },
+              child: Container(
+                margin: const EdgeInsets.all(5),
+                width: double.infinity,
+                padding:
+                    const EdgeInsets.symmetric(horizontal: 20, vertical: 20),
+                child: Text(
+                  "Edit",
+                  style: TextStyle(
+                      color: Colors.white,
+                      fontSize: 20,
+                      fontWeight: FontWeight.bold),
+                  textAlign: TextAlign.center,
+                ),
+                decoration: BoxDecoration(
+                    color: CustomColors.main,
+                    borderRadius: BorderRadius.circular(10)),
+              ),
+            ),
+            GestureDetector(
+              onTap: () {
+                showDialog(
+                  context: context,
+                  builder: ((context) => AlertDialog(
+                        content: Column(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          mainAxisSize: MainAxisSize.min,
+                          children: [
+                            const Text(
+                              "You are about to delete this resource.\nThis action is irreversable.\nDo you want to continue?",
+                              style: TextStyle(
+                                color: Colors.red,
+                                fontWeight: FontWeight.bold,
+                                fontSize: 22,
+                              ),
+                              textAlign: TextAlign.center,
+                            ),
+                            Row(
+                              mainAxisAlignment: MainAxisAlignment.center,
+                              children: [
+                                GestureDetector(
+                                  onTap: () async {
+                                    int? deleted = await contentService
+                                        .deleteResource(resourceId: r.id);
+                                    if (deleted == 201) {
+                                      //TODO: remove the resource tile from here
+                                      ScaffoldMessenger.of(context)
+                                          .showSnackBar(
+                                        const SnackBar(
+                                            content: Text(
+                                                'Resource Successfully Deleted!')),
```

Comment:
Page does not refresh immidiately after deleting a resource. Here is a screenshot after I deleted resource 'How black is your Black' and at the end of the page it says 'Resource Successfully Deleted!', however, it is still visible on the page.

<img src="https://user-images.githubusercontent.com/56507424/205575744-defdd9dc-f5dc-4112-be97-fa08e6aba2de.jpeg" width = 35%>

Reply:
I fixed this problem with my last commit

Reply:
This one also works.